src/@next/AuthenticationInput/AuthenticationInput.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { ClipboardEvent, RefObject, useEffect, useState } from 'react';
+import { StyledInputsContainer } from './AuthenticationInputStyle';
+import { AuthenticationInputSingle } from './AuthenticationInputSingle';
+
+export interface AuthenticationInputProps
+  extends Omit<React.HTMLAttributes<HTMLDivElement>, 'onChange'> {
+  numberOfInputs: number;
+  disabled?: boolean;
+  error?: boolean;
+  forwardedRef?: RefObject<HTMLDivElement>;
+  onChange?: (value: string | null) => void;
+}
+const AuthenticationInputComponent = ({
+  numberOfInputs = 6,
+  error,
+  onChange,
+  forwardedRef,
+  ...rest
+}: AuthenticationInputProps) => {
+  const [inputValues, setInputValues] = useState(
+    new Array<number | ''>(numberOfInputs).fill('')
+  );
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const inputArray = new Array(numberOfInputs).fill('');
+
+  useEffect(() => {
+    if (inputValues.some(v => !Number.isInteger(v))) {
+      onChange?.(null);
+      return;
+    }
+    onChange?.(inputValues.join(''));
+  }, [inputValues, onChange]);
```

Comment:
this onChange could cause infinite looping issue if the callback provided by the user keeps changing its reference on every render? should we fix the same here instead of expecting the user/client to pass a callback which does not change its reference every time?

Reply:
good point, use ref to handle the exhaustive deps warning